M701 and M702 support from LCD and Gcode

q3ok opened the pull request:
From the M600 I've created the M701 and M702 support, and I've added it as new menu items to Prepare menu. There is no translations, its only in en language for display. The filament load are using still the screen from M600 with "Resume print" message, it has to be changed also.

q3ok committed:
Added support for M701 and M702

Added support for filament loading and unloading from prepare menu,
using default settings for filament change (M600) option, which is
required turned on to use the filament load and unload optons.

q3ok committed:
Missing file for M701 and M702

q3ok committed:
M701 and M702 missing file modification from LCD

q3ok committed:
Added babystepping to main LCD menu, added default Configuration for Prusa Clone

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -7318,6 +7320,143 @@ inline void gcode_M503() {
+   */
+  inline void gcode_M702() {
+    
+	  if (thermalManager.tooColdToExtrude(active_extruder)) {
```
Please do a global search and replace to replace tabs with spaces.

q3ok replied:
Global tabs has been replaced with spaces, I believe that i didn't missed any

thinkyhead commented on Marlin/Marlin_main.cpp:
```diff
@@ -7318,6 +7320,143 @@ inline void gcode_M503() {
     lcd_filament_change_show_message(FILAMENT_CHANGE_MESSAGE_STATUS);
   }
 
+    /**
```
Remove extra leading space(s) here.

q3ok replied:
Spaces removed

thinkyhead commented:
Overall the implementation looks good (though I have yet to review it fully). Note that all new features must be made optional and disabled by default, so you'll need to add a new configuration option to make it possible to enable/disable this feature, and set it disabled.

thinkyhead reviewed (changes requested):
Some spacing changes requested…

q3ok committed:
Additional gcodes after lcd print stop

+ added (up to 5) additional gcodes to run after stop print
* modified configuration for MK2Clone, added encoder steps per menu item
* modified buld version with added mk2clone

q3ok committed:
Added flament load/unload as option, fixed tabs

- added FILAMENT_LOADUNLOAD_SUPPORT option in configuration_adv
- replaced all tabs
- removed extra blank lines
- added polish translation for filament load/unload

q3ok commented:
The option FILAMENT_LOADUNLOAD_SUPPORT has been added in Configuration_adv

q3ok committed:
Missing #ifdef

q3ok committed:
Modified missed ifdef check after adding new option in configuration

q3ok committed:
Merge branch 'MK2Clone' into RC

q3ok committed:
Revert "Merge branch 'MK2Clone' into RC"

This reverts commit f577c193ab1871dc17b7b95d780d2f9f241cddbf, reversing
changes made to 14b96f0dc10487e33c774359bf778c0c7cd76398.

tcm0116 commented:
@q3ok I recently submitted a pull request (https://github.com/MarlinFirmware/Marlin/pull/6407) that deals with the Filament Reload and Pause features. We might want to coordinate since this one is related to that group.

q3ok commented:
@tcm0116 I am not sure if even the pause feature should be unified with filament change. IMO that should be two different options, but I can agree that M600 can be splitted into at least two functions, as well to use with M701, M702 and M125.
After few days of using the M701 and M702 made from M600 i found that there is a bug after filament load, which prevents to start any print - I didn't found yet the reason.

tcm0116 commented:
@q3ok By unify, I meant that it would make sense to have them share a set of common functions in the code, but still use the respective M codes. This would prevent having very similar code in several different places that has had small changes made to it.

tcm0116 commented:
@q3ok In my setup, I'm using an E3D Cyclops hotend, and they actually recommend feeding both filaments cold as far as possible before heating up the nozzle. As such, it would be nice to have the option to load the filament with the nozzle cold, and possibly even the option of feeding both extruders simultaneously. This would then require that the hotend be heated up and then the filament loaded the rest of the way. Thoughts on this?

thinkyhead commented:
#7107 supplants this PR…
